src/Disks/ObjectStorages/AzureBlobStorage/AzureBlobStorageCommon.cpp

```diff
@@ -608,31 +608,37 @@ void AzureSettingsByEndpoint::loadFromConfig(
 
     for (const String & key : config_keys)
     {
-        const auto key_path = config_prefix + "." + key;
-        String endpoint_path = key_path + ".connection_string";
+        if (config.has(config_prefix + "." + key + ".object_storage_type")) {
+            const auto &object_storage_type = config.getString(config_prefix + "." + key + ".object_storage_type");
+            if (object_storage_type.find("azure") == std::string::npos)
+            {
+                /// Then its not an azure config
+                continue;
```

Comment:
@Kseniia : We can have both `azure` & `azure_blob_storage` for `object_storage_type` in config is it ? I see both used, just wanted to confirm